### PR TITLE
Add Speaker Diarization doc 

### PR DESCRIPTION
**Pull request by fayejf**
cleanup [previous PR](https://github.com/NVIDIA/NeMo/pull/1928)

### CODE REVIEW

File: docs/source/asr/speaker_diarization/intro.rst [hunk 1]
```diff
@@ -0,0 +1,70 @@
+Speaker Diarization
+==================================
+
+Speaker Diarization (SD) is the task of segmenting audio recordings by speaker labels, that is Who Speaks When?
+
+.. image:: images/sd_pipeline.png
```
Comment (titu1994) [inline docs/source/asr/speaker_diarization/intro.rst:6]: The image looks great, but it has black text on black background (or maybe thats just my browser setting under dark mode). Just a heads up.

File: docs/source/asr/speaker_diarization/intro.rst [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        :align: center
+        :alt: Speaker Diarization pipeline [todo]
+        :scale: 50%
+
+
+A diarization system consists of **Voice Activity Detection (VAD)** model to get the time stamps of audio where speech is being spoken ignoring the background and
```
Comment (titu1994) [inline docs/source/asr/speaker_diarization/intro.rst:12]: ... being spoken while ignoring the ...

File: docs/source/asr/speaker_diarization/intro.rst [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        
+Run and evaluate speaker diarizer with below command:
+
+.. code-block:: bash
+
+  python examples/speaker_recognition/speaker_diarize.py \
+  diarizer.paths2audio_files=<either list of audio file paths or file containing paths to audio files> \
+  diarizer.path2groundtruth_rttm_files=<(Optional) either list of rttm file paths or file containing paths to rttm files> \
+  diarizer.vad.model_path="vad_telephony_marblenet" \
+  diarizer.vad.threshold=0.7 \
```
Comment (titu1994) [inline docs/source/asr/speaker_diarization/intro.rst:32]: This is a duplicate of your speaker_diarizer.py docstring no? Cant we just refer to it and avoid this. Future updates will break this codeblock - so its double work to maintain it.

**Review by titu1994 — CHANGES_REQUESTED** [review]
Overall looks great ! Minor comments, ignore the ones which are just heads up

File: docs/source/asr/speaker_diarization/configs.rst [hunk 4]
```diff
@@ -0,0 +1,65 @@
+NeMo Speaker Diarization Configuration Files
+============================================
+
+Since speaker diarization model here is not a fully-trainble End-to-End model but a inference pipeline, we use **diarizer** instead of **model** which is used in other tasks.
```
[the remark this answers is not in the document]
Comment (fayejf) [reply docs/source/asr/speaker_diarization/configs.rst:4]: yes let's leave it for now. We will have model-like sd model later

File: docs/source/asr/speaker_diarization/intro.rst (hunk 1 above)
Comment (fayejf) [reply docs/source/asr/speaker_diarization/intro.rst:6]: It looks fine under dark model in my chrome though ugly...

File: docs/source/asr/speaker_diarization/intro.rst (hunk 3 above)
Comment (fayejf) [reply docs/source/asr/speaker_diarization/intro.rst:32]: ouch, we don't have good instruction in speaker_diarize.py 
@nithinraok  shall we update it and move it?

File: docs/source/asr/speaker_diarization/intro.rst [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+Speaker Diarization (SD) is the task of segmenting audio recordings by speaker labels, that is Who Speaks When?
+
+.. image:: images/sd_pipeline.png
+        :align: center
+        :alt: Speaker Diarization pipeline [todo]
+        :scale: 50%
+
+
+A diarization system consists of **Voice Activity Detection (VAD)** model to get the time stamps of audio where speech is being spoken ignoring the background and 
+**Speaker Embeddings** model to get speaker embeddings on segments that were previously time stamped by VAD.
```
Comment (jbalam-nv) [inline docs/source/asr/speaker_diarization/intro.rst:13]: **Speaker Embeddings** model -> a **Speaker Embeddings** model
 on segments that were previously time stamped by VAD ->  on speech segments obtained  from VAD time stamps.

File: docs/source/asr/speaker_diarization/models.rst [hunk 6]
```diff
@@ -0,0 +1,5 @@
+Models
+======
+
+
+Currenlty NeMo's Speaker Diarization pipeline use `MarbleNet <../speech_classification/models.html#marblenet-vad>`__ model for Voice Activity Detection (VAD) and `SpeakerNet <../speaker_recognition/models.html#speakernet>`__ model for speaker embedding extraction.
```
Comment (jbalam-nv) [inline docs/source/asr/speaker_diarization/models.rst:5]: pipeline use -> pipeline uses

File: docs/source/asr/speaker_diarization/results.rst [hunk 7]
```diff
@@ -0,0 +1,61 @@
+Checkpoints
+===========
+
+There are two main ways to load pretrained checkpoints in NeMo as introduced in `loading ASR checkpoints <../results.html#checkpoints>`__.
+In speaker diarization, the diarizer would take care of model loading, you can load checkpoints by passing it thourgh config file. For example:
```
Comment (jbalam-nv) [inline docs/source/asr/speaker_diarization/results.rst:5]: In speaker diarization, the diarizer would take care of model loading, you can load checkpoints by passing it thourgh config file -> In speaker diarization, the diarizer  loads checkpoints that are passed through the config file

**Review by jbalam-nv — APPROVED** [review]
Minor changes to grammar. Great work @fayejf and @nithinraok

File: docs/source/asr/speaker_diarization/intro.rst (hunk 3 above)
Comment (nithinraok) [reply docs/source/asr/speaker_diarization/intro.rst:32]: Move it, we shall update the function call and comeback to this again after thats done

File: docs/source/asr/speaker_diarization/intro.rst (hunk 1 above)
Comment (titu1994) [reply docs/source/asr/speaker_diarization/intro.rst:6]: Yup darkmode is the issue, light mode seems fine.

Comment (nithinraok) [reply docs/source/asr/speaker_diarization/intro.rst:6]: We are not supporting dark mode right in nemo docs? browser settings dark mode are hacks IMO

**Review by titu1994 — APPROVED** [review]
Looks awesome ! Merge when ready

**Pull request merged**